src/platform/packages/private/kbn-scout-reporting/src/registry/test_config.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { globSync } from 'fast-glob';
+import { REPO_ROOT } from '@kbn/repo-info';
+import path from 'node:path';
+import { ToolingLog } from '@kbn/tooling-log';
+import { SCOUT_CONFIG_PATH_GLOB, SCOUT_CONFIG_PATH_REGEX } from '@kbn/scout-info';
+import { existsSync, readFileSync } from 'node:fs';
+import type { ScoutTestableComponent } from './testable_component';
+import type { ScoutConfigManifest } from './manifest';
+
+export interface ScoutTestConfig {
+  path: string;
+  category: string;
+  type: string;
+  component: ScoutTestableComponent;
+  manifest: ScoutConfigManifest;
+}
+
+export const testConfig = {
+  fromPath(configPath: string): ScoutTestConfig {
+    const match = configPath.match(SCOUT_CONFIG_PATH_REGEX);
+
+    if (match == null) {
+      throw new Error(`Scout config path ${configPath} did not match the expected regex pattern`);
+    }
+
+    const [
+      _,
+      platform,
+      solution,
+      componentType,
+      componentVisibility,
+      componentName,
+      customTargetConfigSetName,
+      testCategory,
+      testConfigType,
+    ] = match;
+
+    const scoutDirName = `scout${customTargetConfigSetName ? `_${customTargetConfigSetName}` : ''}`;
+
+    const component = {
+      name: componentName,
+      group: platform ?? solution,
+      type: componentType.slice(0, -1) as ScoutTestableComponent['type'],
+      visibility: componentVisibility as ScoutTestableComponent['visibility'],
+      root: configPath.split(`/test/${scoutDirName}`)[0],
+    };
+
+    const manifestPath = path.join(
+      component.root,
+      'test',
+      scoutDirName,
+      '.meta',
+      testCategory,
+      `${testConfigType || 'standard'}.json`
+    );
+    const manifestExists = existsSync(manifestPath);
+
+    const manifest: ScoutConfigManifest = manifestExists
+      ? {
+          path: manifestPath,
+          exists: manifestExists,
+          ...JSON.parse(readFileSync(manifestPath, 'utf8')),
```

Comment:
_question_: can me move it above and validate if it is a JSON? just in case we got "corrupted", would good to see a nice error.